src/prezzemolo/graph.py

```diff
@@ -87,6 +110,42 @@ def breadth_first_search(self, start: Vertex[ValueType], end: Vertex[ValueType])
         # start and end are type checked inside _breadth_first_search()
         return self._breadth_first_search(start, end, vertex_2_parent=None)
 
+    def _dijkstra_search(
+        self, start: Vertex[ValueType], end: Optional[Vertex[ValueType]], vertex_2_parent: Optional[Dict[Vertex[ValueType], Optional[Vertex[ValueType]]]] = None
+    ) -> bool:
+        distance: Dict[Vertex[ValueType], float] = {v: float("inf") for v in self.__vertexes}
+        distance[start] = 0.0
+        remaining_vertexes: PriorityQueue[_ShortestDistance[ValueType]] = PriorityQueue()
+        remaining_vertexes.put(_ShortestDistance(distance=0.0, vertex=start))
+        visited: Set[Vertex[ValueType]] = set()
+
+        while not remaining_vertexes.empty():
+            next_vertex: _ShortestDistance[ValueType] = remaining_vertexes.get()
+            current_distance: float = next_vertex.distance
+            current_vertex: Vertex[ValueType] = next_vertex.vertex
+
+            if current_vertex in visited:
+                continue
+
+            visited.add(current_vertex)
+
+            for neighbor in current_vertex.neighbors:
+                neighbor_distance = current_distance + current_vertex.get_weight(neighbor)
+                if neighbor_distance < distance[neighbor]:
+                    if vertex_2_parent is not None:
+                        vertex_2_parent[neighbor] = current_vertex
+                    distance[neighbor] = neighbor_distance
+                    remaining_vertexes.put(_ShortestDistance(distance=neighbor_distance, vertex=neighbor))
+
+            if current_vertex == end:
+                return True
+
+        return False
+
+    def dijkstra_search(self, start: Vertex[ValueType], end: Vertex[ValueType]) -> bool:
```

Comment:
I'm thinking we probably don't need this: it doesn't offer any advantage over `breadth_first_search()`: both these functions just check if the two vertices are connected. I'm also thinking we should rename the following:
* `breadth_first_search()` -> `are_connected()`
* `find_shortest_path()` -> `breadth_first_search()`
* `find_weighted_shortest_path()` -> `dijkstra()` 
* `_dijkstra_search()` -> `_dijkstra()` (there is no need to qualify it with "search": in CS when you say Dijkstra, it's understood to be his path search algorithm)